meson.build

```diff
@@ -0,0 +1,304 @@
+#
+# Old configure.ac
+#
+
+project(
+  'nix',
+  'cpp',
+  default_options : [
+    'cpp_std=c++14',
+    #'localstatedir=/nix/var',
+  ],
+)
+
+sed = find_program('sed')
+
+conf_data = configuration_data()
+
+system = get_option('with-system')
+if system == ''
+  system = host_machine.cpu() + '-' + host_machine.system()
+endif
+conf_data.set_quoted('SYSTEM', system, description : 'platform identifier (`cpu-os\')')
+
+# State should be stored in /nix/var, unless the user overrides it explicitly.
+if get_option('normal-var')
```

Comment:
Can we determine if an option has been manually set by the user in Meson? Adding an option like this is confusing.

Reply:
It is confusing; I forget where I got stuck before. This is the autoconf original https://github.com/NixOS/nix/blob/master/configure.ac#L49-L50 . Nix hijack's the variable, in a way. I think the issue was you cannot force-set such a thing within `meson.build`, but you should still be able to compare the user-given value to `${prefix}/var` like the autoconf. That prevents the user from forcing it to be `${prefix}/var`, but hey that's no worse than today.